packages/eslint-plugin/src/rules/no-magic-numbers.ts

```diff
@@ -0,0 +1,149 @@
+/**
+ * @fileoverview Rule to flag statements that use magic numbers (adapted from https://github.com/danielstjules/buddy.js)
+ * @author Scott O'Hara
+ */
+
+import { TSESTree, AST_NODE_TYPES } from '@typescript-eslint/typescript-estree';
+import baseRule from 'eslint/lib/rules/no-magic-numbers';
+import * as util from '../util';
+import { JSONSchema4 } from 'json-schema';
+
+type Options = util.InferOptionsTypeFromRule<typeof baseRule>;
+type MessageIds = util.InferMessageIdsTypeFromRule<typeof baseRule>;
+
+// Original schema properties from the base rule
+const { properties } = (baseRule.meta.schema as JSONSchema4[])[0];
+
+// Extend base schema with additional property to ignore TS numeric literal types
+if (properties) {
+  properties.ignoreNumericLiteralTypes = {
+    type: 'boolean',
+  };
+}
+
+export default util.createRule<Options, MessageIds>({
+  name: 'no-magic-numbers',
+  meta: {
+    type: 'suggestion',
+    docs: {
+      description: 'Disallow magic numbers',
+      category: 'Best Practices',
+      recommended: false,
+    },
+    schema: baseRule.meta.schema,
+    messages: baseRule.meta.messages,
+  },
+  defaultOptions: [
+    {
+      ignore: [],
+      ignoreArrayIndexes: false,
+      enforceConst: false,
+      detectObjects: false,
+      ignoreNumericLiteralTypes: false,
```

Comment:
do we want it to be true by default for our rule?
No preference, just putting it up for discussion

Reply:
It is [not recommended for eslint](https://eslint.org/docs/rules/), so I'd say no.

Reply:
I think perhaps @bradzacher you're asking whether `ignoreNumericLiteralTypes` should default to `true`; and @ThomasdenH you're indicating that the rule itself should not be a recommended rule?

I have no strongly held views either way, but my personal opinions are:

- I would advocate for `ignoreNumericLiteralTypes` to default to `true` (happy to change from the current default of `false` if there is consensus to do so)
- agree that this should not be a recommended rule (I've updated the `README` and removed the ✔️ that was incorrectly added)